components/ContributionConfirmationModal.js

```diff
@@ -0,0 +1,237 @@
+      paymentProcessorFeesAmount: { valueInCents: paymentProcessorFee, currency },
+      platformTipAmount: { valueInCents: platformTip, currency },
+    };
+    try {
+      await confirmOrder({ variables: { id: order.id, action: 'MARK_AS_PAID', details: orderConfirmationDetails } });
+      addToast({
+        type: TOAST_TYPE.SUCCESS,
+        message: intl.formatMessage({ defaultMessage: 'Order confirmed successfully' }),
+      });
+      onClose();
+    } catch (e) {
+      addToast({ type: TOAST_TYPE.ERROR, message: i18nGraphqlException(intl, e) });
+    }
+  };
+
+  const handleChange = (fieldName, value) => {
+    if (fieldName === 'amountReceived') {
+      setAmountReceived(value);
+    } else if (fieldName === 'platformTip') {
+      setPlatformTip(value);
+    } else if (fieldName === 'paymentProcessorFee') {
+      setPaymentProcessorFee(value);
+    }
+  };
+
+  return (
+    <Modal width="578px" show={show} onClose={onClose} trapFocus>
+      <ModalHeader>
+        <FormattedMessage
+          defaultMessage="Confirm contribution to {collectiveName}"
+          values={{
+            collectiveName: order.toAccount.name,
+          }}
+        />
+      </ModalHeader>
+      <ModalBody>
+        <P mb={4} fontSize="13px">
+          <FormattedMessage defaultMessage="Confirm the amount of funds you have received in your host account." />
+        </P>
+        <Container mt="58px">
+          <Flex justifyContent="space-between" alignItems={['left', 'center']} flexDirection={['column', 'row']}>
+            <Span fontSize="14px" lineHeight="20px" fontWeight="700">
+              <FormattedMessage
+                defaultMessage="Amount initiated by {contributor}"
+                values={{ contributor: order.fromAccount.name }}
+              />
+            </Span>
+            <Box width="142px">
+              <InputField
+                className="horizontal"
+                name="amountInitiated"
+                type="currency"
+                onChange={() => null}
+                pre={getCurrencySymbol(currency)}
+                value={amountInitiated}
+                disabled
+              />
+            </Box>
+          </Flex>
+        </Container>
+        <StyledHr borderStyle="solid" mt="16px" mb="16px" />
+        <Container>
+          <Flex justifyContent="space-between" alignItems={['left', 'center']} flexDirection={['column', 'row']}>
+            <Span fontSize="14px" lineHeight="20px" fontWeight="400">
+              <FormattedMessage
+                defaultMessage="Amount received by {collective}"
+                values={{ collective: order.toAccount.name }}
+              />
+            </Span>
+            <Box width="142px" data-cy="amount-received">
+              <InputField
+                className="horizontal"
+                name="amountReceived"
+                type="currency"
+                onChange={value => handleChange('amountReceived', value)}
```

Comment:
From my point of view, the `handleChange` helper doesn't simplify nor clarify the code since we then do a `if` / `else if` on the field name. I think dispatching the change directly is a clean option:

```suggestion
                onChange={value => setAmountReceived(value)}
```

Reply:
ops, I've corrected it. 👍🏽